### DISCUSSION TITLE
Java for desktop Cruiser

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#132] [Q&A] [answered]
Want to try the desktop version. Which Java Runtime is required, 32 Bit or 64 Bit?

_Originally posted by @DolomitesTom in https://github.com/devemux86/cruiser/discussions/40#discussioncomment-5681951_

**Comment by devemux86** [comment, marked as answer]
These days the default should be 64 bit and preferably a long term version:
https://adoptium.net/temurin/releases/?version=11

On desktop the "Mapsforge" rendering engine may work better,
as "VTM (OpenGL)" depends on the operating system and graphics drivers.

**Reply by DolomitesTom** [reply]
OK thanks will try it.

**Comment by devemux86** [comment]
For keyboard shortcuts and installation by operating system, see the Cruiser page on OpenStreetMap Wiki:

- https://wiki.openstreetmap.org/wiki/Cruiser

**Comment by DolomitesTom** [comment]
Cruiser for Desktop looks very nice, really good for planning. 

It started to work, but after switching to Open GL it seems to be stuck in an endless loop, takes almost 1 minute to open. I have loaded and open the maps (France, Germany, Luxemburg, Italy, Belgium). Then I tried several menus and suddenly it crashed when I clicked on Settings. All Settings got lost. They obviously only get saved when exit the program. 
Question: how can I hide the yellow turn instructions?

**Reply by devemux86** [reply]
> Question: how can I hide the yellow turn instructions?

There are no such extensive configuration options as in Android Cruiser.
Maybe I can add it in the future, e.g. via a keyboard shortcut?

Desktop Cruiser started a project to help map / theme creators, as testing on desktop is much easier than on Android.

**Comment by devemux86** [comment]
OpenGL on desktop (like games) depends on the operating system and graphics drivers.

I work on Linux, so I don't usually get the chance to test on Windows or macOS very often. :slightly_smiling_face: 

---

You can restart Cruiser with Mapsforge with the command:

```console
java -jar cruiser.jar mapsforge
```

**Comment by DolomitesTom** [comment]
Unfortunately after working a while with it quite good, it stucked again after inserting a via point and renaming it.

**Reply by devemux86** [reply]
Do you use the _**recommended**_ Mapsforge version?

Is it about route calculation? What routing service?
Try without avoid options.

Can you post a screenshot?

**Comment by devemux86** [comment]
If you are using **desktop** Cruiser for routing, it is better to use the routing sites in the browser.
They usually provide more routing options and are more comfortable to use on their websites.

For example, BRouter website allows direct editing of routing profiles.

**Comment by devemux86** [comment]
@DolomitesTom did you have too many favorites in desktop Cruiser?

Opening desktop Cruiser (OpenGL) with too many favorites (thousands) was slow.
It will work better in the next desktop Cruiser update.

**Reply by DolomitesTom** [reply]
Yes all Alpes paces and much more.

**Reply by devemux86** [reply]
@DolomitesTom thanks for the confirmation!

Next desktop Cruiser will save / load favorites differently and **instantly**.

This will work after the first start and the conversion to the new format.
So until then, it is better to avoid large numbers of favorites.

_And on desktop use Mapsforge as rendering engine, instead of VTM (OpenGL)._

**Reply by devemux86** [reply]
> Next desktop Cruiser will save / load favorites differently and instantly.

Forgot to mention this is already available.

Favorites are now stored in a gpx file inside the desktop app folder.
So they are instantly loaded when Cruiser starts.
